### PR TITLE
Transforms implementation and refactor

### PR DESCRIPTION
**Pull request by ErikBjare**
aw-server will need it's equivalent PR merged simultaneously.

### CODE REVIEW

File: aw_core/transforms.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # Event doesn't overlap
+    events = [Event(timestamp=datetime.now(), duration=timedelta(hours=1))]
+    afkevents = [Event(timestamp=datetime.now() + timedelta(hours=2), duration=timedelta(hours=1))]
+    print(filter_afk_events(events, afkevents))
+"""
+
+
+def chunk(events: List[Event]) -> dict:
+    eventcount = 0
+    chunk = dict()  # type: Dict[str, Any]
```
Comment (johan-bjareholt) [inline aw_core/transforms.py]: I think i wrote this, but this variable should be "chunks" instead of "chunk"

File: aw_core/transforms.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    print(filter_afk_events(events, afkevents))
+
+    # Event doesn't overlap
+    events = [Event(timestamp=datetime.now(), duration=timedelta(hours=1))]
+    afkevents = [Event(timestamp=datetime.now() + timedelta(hours=2), duration=timedelta(hours=1))]
+    print(filter_afk_events(events, afkevents))
+"""
+
+
+def chunk(events: List[Event]) -> dict:
```
Comment (johan-bjareholt) [inline aw_core/transforms.py]: I wrote this heh, but it's borderline complex so I should probably have documented it better

Comment (ErikBjare) [inline aw_core/transforms.py]: It seems to work great so I'm not touching it :smile:

File: aw_core/transforms.py (hunk 1 above)
Comment (ErikBjare) [inline aw_core/transforms.py]: Fixed!

**Comment by ErikBjare** [comment]
@johan-bjareholt Alright, so here is a hopefully-working version of the difficult-to-name `filter_period_intersect`. Give it a go and let me know how it works. You need to filter away the AFK events so you only have not-AFK events left and then pass those as the second argument.

**Pull request merged**